Form/ChoiceList/PhpcrOdmQueryBuilderLoader.php

```diff
@@ -77,10 +80,16 @@ public function getEntities()
      */
     public function getEntitiesByIds($identifier, array $values)
     {
+        /* performance: if we could figure out whether the query builder is "
+         * empty" (that is only checking for the class) we could optimize this
+         * to a $this->dm->findMany(null, $values)
+         */
+
         $qb = clone $this->queryBuilder;
         $alias = $qb->getPrimaryAlias();
+        $where = $qb->andWhere()->orX();
         foreach ($values as $val) {
-            $qb->orWhere()->eq()->field($alias.'.'.$identifier)->literal($val);
+            $where->same($val, $alias);
```

Comment:
docs seem to agree as well http://doctrine-phpcr-odm.readthedocs.org/en/latest/reference/query-builder-reference.html#qbref-method-constraintfactory-orx